src/gas/determineGasFeeSuggestions.test.ts

```diff
@@ -0,0 +1,294 @@
+          throw new Error('Some API failure');
+        });
+      });
+
+      describe('assuming fetchEthGasPriceEstimate does not throw an error', () => {
+        it('returns the fetched fee estimates and an empty set of time estimates', async () => {
+          const gasFeeEstimates = buildMockDataForFetchEthGasPriceEstimate();
+          fetchEthGasPriceEstimate.mockResolvedValue(gasFeeEstimates);
+
+          const gasFeeSuggestions = await determineGasFeeSuggestions(options);
+
+          expect(gasFeeSuggestions).toStrictEqual({
+            gasFeeEstimates,
+            estimatedGasFeeTimeBounds: {},
+            gasEstimateType: 'eth_gasPrice',
+          });
+        });
+      });
+
+      describe('when fetchEthGasPriceEstimate throws an error', () => {
+        it('throws an error that wraps that error', async () => {
+          fetchEthGasPriceEstimate.mockImplementation(() => {
+            throw new Error('fetchEthGasPriceEstimate failed');
+          });
+
+          const promise = determineGasFeeSuggestions(options);
+
+          await expect(promise).rejects.toThrow(
+            'Gas fee/price estimation failed. Message: fetchEthGasPriceEstimate failed',
+          );
+        });
+      });
+    });
+  });
+
+  describe('when isEIP1559Compatible is false but isLegacyGasAPICompatible is true', () => {
+    beforeEach(() => {
+      options = {
+        isEIP1559Compatible: false,
+        isLegacyGasAPICompatible: true,
+        fetchGasEstimates,
+        fetchGasEstimatesUrl: 'http://doesnt-matter',
+        fetchLegacyGasPriceEstimates,
+        fetchLegacyGasPriceEstimatesUrl:
+          'http://some-legacy-gas-price-estimates-url',
+        fetchEthGasPriceEstimate,
+        calculateTimeEstimate,
+        clientId: 'some-client-id',
+        ethQuery: {},
+      };
+    });
+
+    describe('assuming fetchLegacyGasPriceEstimates does not throw an error', () => {
+      it('returns the fetched fee estimates and an empty set of time estimates', async () => {
+        const gasFeeEstimates = buildMockDataForFetchLegacyGasPriceEstimates();
+        fetchLegacyGasPriceEstimates.mockResolvedValue(gasFeeEstimates);
+
+        const gasFeeSuggestions = await determineGasFeeSuggestions(options);
+
+        expect(gasFeeSuggestions).toStrictEqual({
+          gasFeeEstimates,
+          estimatedGasFeeTimeBounds: {},
+          gasEstimateType: 'legacy',
+        });
+      });
+    });
+
+    describe('when fetchLegacyGasPriceEstimates throws an error', () => {
+      beforeEach(() => {
+        fetchLegacyGasPriceEstimates.mockImplementation(() => {
+          throw new Error('Some API failure');
+        });
+      });
+
+      describe('assuming fetchEthGasPriceEstimate does not throw an error', () => {
+        it('returns the fetched fee estimates and an empty set of time estimates', async () => {
+          const gasFeeEstimates = buildMockDataForFetchEthGasPriceEstimate();
```

Comment:
I could extract the return value there to a variable and reuse it here, but I tend to have a philosophy when testing that data that is important to the test should be specified in the test itself and not outside. The fact that this is also in `beforeEach` is somewhat of a coincidence in order to make TypeScript happy when stubbing `fetchEthGasPriceEstimate` there (otherwise it complains that the function needs to return something that fits the shape it expects). So although they look the same, they are independent of each other so that either can be changed without breaking this test. That said, this is a style thing and I'm curious to know what would make sense to you and how you would write this. I suppose one way to make this less confusing would be to remove `beforeEach` and only stub the functions we care about in the tests themselves. What do you think?